src/auth/auth.swagger.ts

```diff
@@ -1,58 +1,165 @@
+import { applyDecorators } from '@nestjs/common';
+import { ApiOperation, ApiBody, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { LoginRequestDto } from './dto/login-request.dto';
 import { SignupRequestDto } from './dto/signup-request.dto';
+import { AuthResponseDto } from './dto/auth-response.dto';
 
-export const signupSwagger = {
-  apiBody: {
-    description: '회원가입 요청 DTO',
-    type: SignupRequestDto,
-  },
-  apiCreatedResponse: {
-    description: '회원가입 성공',
-    schema: {
-      example: {
-        status: 'success',
-        message: '회원가입에 성공했습니다.',
-        data: null,
-      },
-    },
+const unauthorizedExamples = {
+  TokenExpired: {
+    message: '토큰이 만료되었습니다. 다시 로그인해주세요.',
+    error: 'Unauthorized',
+    statusCode: 401,
   },
-  apiBadRequestResponse: {
-    description: '회원가입 실패',
-    schema: {
-      example: {
-        message: '이미 사용 중인 사용자 ID입니다.',
-        error: 'Bad Request',
-        statusCode: 400,
-      },
-    },
+  InvalidToken: {
+    message: '유효하지 않은 토큰입니다.',
+    error: 'Unauthorized',
+    statusCode: 401,
+  },
+};
+
+const conflictExamples = {
+  DuplicateUser: {
+    message: '이미 사용 중인 사용자 ID입니다.',
+    error: 'Conflict',
+    statusCode: 409,
   },
 };
 
-export const loginSwagger = {
-  apiBody: {
-    description: '로그인 요청 DTO',
-    type: LoginRequestDto,
-  },
-  apiCreatedResponse: {
-    description: '로그인 성공',
-    schema: {
-      example: {
-        status: 'success',
-        message: '로그인에 성공했습니다.',
-        data: {
-          accessToken: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ...',
+const badRequestExamples = {
+  PasswordPattern: {
+    message: ['비밀번호는 영문, 숫자, 특수문자를 모두 포함해야 합니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+  EmailPattern: {
+    message: ['이메일 형식이 올바르지 않습니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+  Required: {
+    message: ['필수 값이 누락되었거나 형식이 올바르지 않습니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+  LoginFail: {
+    message: ['아이디 또는 비밀번호가 틀렸습니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+  RefreshToken: {
+    message: ['Refresh Token이 없습니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+  UserNotFound: {
+    message: ['존재하지 않는 사용자입니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+  InvalidRefreshToken: {
+    message: ['유효하지 않은 Refresh Token입니다.'],
+    error: 'Bad Request',
+    statusCode: 400,
+  },
+};
+
+const unauthorizedResponses = () =>
+  ApiResponse({
+    status: 401,
+    description: '인증되지 않음 (JWT 토큰 없음, 만료, 유효하지 않음)',
+    content: {
+      'application/json': {
+        examples: {
+          '토큰 만료': { value: unauthorizedExamples.TokenExpired },
+          '유효하지 않은 토큰': { value: unauthorizedExamples.InvalidToken },
```

Comment:
좀 오래돼서 기억이 정확히지는 않은데요 examples가 기본적으로 application/json으로 만들어주는거로 기억해요

```suggestion
        examples: {
          '토큰 만료': { value: unauthorizedExamples.TokenExpired },
          '유효하지 않은 토큰': { value: unauthorizedExamples.InvalidToken },
```